executor/writetest/write_test.go

```diff
@@ -592,6 +592,35 @@ commit;`
 	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1526 Table has no partition for value 3"))
 }
 
+func TestIssue38950(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	var cfg kv.InjectionConfig
+	tk := testkit.NewTestKit(t, kv.NewInjectedStore(store, &cfg))
+	tk.MustExec("use test")
+	testSQL := `drop table if exists t;
+    create table t (id smallint auto_increment primary key);`
+	tk.MustExec(testSQL)
+
+	testSQL = `alter table t add column c1 int default 1;`
+	tk.MustExec(testSQL)
+
+	testSQL = `insert ignore into t(id) values (194626268);`
+	tk.MustExec(testSQL)
```

Comment:
Minor, keep as-is if you prefer, otherwise I would suggest to have the testSQL string directly in the `tk.MustExec()`, like
```
tk.MustExec("alter table t add column c1 int default 1")
```